activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -566,6 +566,37 @@ def test_expires_in_and_expires_at
     assert_equal "Either :expires_in or :expires_at can be supplied, but not both", error.message
   end
 
+  def test_invalid_expiration_time_raises_an_error_when_raise_on_invalid_cache_expiration_time_is_true
+    ActiveSupport::Cache::Store.raise_on_invalid_cache_expiration_time = true
+
+    freeze_time
+    key = SecureRandom.uuid
+    error = assert_raises(ArgumentError) do
+      @cache.write(key, "bar", expires_at: 1.minute.ago)
+    end
+
+    assert_equal "Cache expiration time is invalid, cannot be negative: -60.0", error.message
+  end
+
+  def test_invalid_expiration_time_reports_and_logs_when_raise_on_invalid_cache_expiration_time_is_false
+    ActiveSupport::Cache::Store.raise_on_invalid_cache_expiration_time = false
+
+    ActiveSupport.error_reporter = ActiveSupport::ErrorReporter.new
```

Comment:
The reason I suggested a block based helper, is that here you are leaking a global value. 

This `ErrorReporter` instance will replace whatever `ActiveSupport.error_reporter` was set to, and survive past this test.

With a block based helper, it's much easier to record the previous value and properly set it back in an `ensure`.